refactor: unify styling

magicmatatjahu opened the pull request:
<!--   Thank you for your contribution. Before you submit the pull request:
1. Follow contributing guidelines, the recommended Git workflow, and any related documentation.
2. Test your changes and attach their results to the pull request.
3. Update the relevant documentation.
-->

**Description**

Changes proposed in this pull request:

Unify styling in component with HTML-Templates styling:
- styling `Info`, `Server`, `Operation`, `Message`, `Schema` components
- the number of lines is related with new dependencies like `postcss`, `tailwind` etc. The rest of the code is converting inline styles/classes into mixed classes.

> In next PR will be styling for Examples, Sidebar and for mobiles.

Examples:

- Info:
![image](https://user-images.githubusercontent.com/20404945/114172103-85151580-9935-11eb-885c-c15d4d1d4d61.png)

- Server 
![image](https://user-images.githubusercontent.com/20404945/114172130-8f371400-9935-11eb-89bb-49a72aa85adb.png)

- Operation 
![image](https://user-images.githubusercontent.com/20404945/114172180-9eb65d00-9935-11eb-85bd-8fdf2f8fd72f.png)

- Message 
![image](https://user-images.githubusercontent.com/20404945/114172229-ad047900-9935-11eb-9af4-26ff3c684bc9.png)

**Related issue(s)**
Part of https://github.com/asyncapi/shape-up-process/issues/86

fmvilas commented on library/src/styles/styles.css:
```diff
@@ -0,0 +1,371 @@
+  .ai-info__links-item {
+    @apply inline-block;
+  }
+  .ai-info__links-item > a {
```
If we're going to choose BEM over utility-first directly on the component, we should strictly avoid using CSS operators like `>`. They will become a problem if someone wants to override the styles applied here as our CSS selector will always have more specificity than theirs.

And honestly, I'd use `@apply` only for those highly repeated styles (like buttons, links, etc.). I'd keep the styles in the component files otherwise.

magicmatatjahu replied:
> If we're going to choose BEM over utility-first directly on the component, we should strictly avoid using CSS operators like >. They will become a problem if someone wants to override the styles applied here as our CSS selector will always have more specificity than theirs.

I agree with it partially 😄 People will also be able to override styles by `ai-info__links-item > a`. Funny fact: Tailwind also uses operators in some classes, especially in `display` etc.

> And honestly, I'd use @apply only for those highly repeated styles (like buttons, links, etc.). I'd keep the styles in the component files otherwise.

And then people won't be able to override styles for particular parts :)

fmvilas replied:
> People will also be able to override styles by `ai-info__links-item > a`

Have you ever tried overriding Bootstrap styles? What do you think is the purpose of BEM? 😄 By doing this, you're breaking BEM and its benefits.

> And then people won't be able to override styles for particular parts :) I based styles on `@apply` directive having this problem in mind.

We don't want people to fully customize the components and all the CSS (unless they override CSS which is always possible). If someone wants to create a fully customized component, they'll be able to replace our components: https://github.com/fmvilas/asyncapi-pitches/blob/master/ui-library.md#customization. Not a problem to solve now.

In my experience, we should go the utility-first approach and maybe use some names for the colors, like primary, secondary, link, text, etc. Then people will be able to customize these colors. The same applies to sizes, paddings, margins, etc. It is, we must create a design system. We don't create CSS classes for the components but for the system the components are adhereing to.

magicmatatjahu replied:
> We don't want people to fully customize the components and all the CSS (unless they override CSS which is always possible). If someone wants to create a fully customized component, they'll be able to replace our components: https://github.com/fmvilas/asyncapi-pitches/blob/master/ui-library.md#customization. Not a problem to solve now.

I go to link and read: `It should be easy to customize the components. Not just at CSS level but also at markup level.` 😄 I'm just kidding. 

@fmvilas I like the idea of ​​utility classes for buttons, borders, etc., and even today in the morning I noticed the problem that there are too many classes. The question is whether to start writing these classes, already or wait for the next cycle?

fmvilas replied:
I'd not move these classes to the CSS file in this cycle. It's a strong decision that will affect customization. In future cycles, we'll work on customization and I think it's then when we have to make a choice.

magicmatatjahu replied:
I will remember! 😄 Style customization is most important thing after component unification. We talked about it under the main task of the cycle, but I found that not only Kyma has his own styles, but also people from Spotify or one person from https://github.com/asyncapi/asyncapi-react/issues/265 issue. The old component has css theming feature and we break it now.

fmvilas replied:
I don't like the styles. Not because of your changes but because the current design of the HTML template is broken and feels patched 😄 So yeah, I like the changes you did 😄

magicmatatjahu commented:
@fmvilas I moved the styles to components files :) Could you check again? I also moved those that are repeated, such as for buttons, because you are right that it is better to think about theming in the next cycles.

fmvilas reviewed (approved):
Looking great 👍 Thanks for taking my opinion into consideration! 🙏
